Library/OcDebugLogLib/OcLog.c

```diff
@@ -140,18 +146,138 @@ GetLogPath (
   return LogPath;
 }
 
+STATIC
 EFI_STATUS
-EFIAPI
-OcLogAddEntry  (
-  IN OC_LOG_PROTOCOL    *OcLog,
-  IN UINTN              ErrorLevel,
-  IN CONST CHAR8        *FormatString,
-  IN VA_LIST            Marker
+GetLogPrefix (
+  IN   CONST CHAR8  *FormatString,
+  OUT  CHAR8        *Prefix
   )
 {
-  EFI_STATUS                  Status;
+  UINTN    MaxLength;
+  UINTN    Index;
+  CHAR8    Curr;
 
-  OC_LOG_PRIVATE_DATA         *Private;
+  ASSERT (FormatString != NULL);
+  ASSERT (Prefix != NULL);
+
+  //
+  // If FormatString just starts with colon, it must be illegal.
+  //
+  if (*FormatString == ':') {
+    return EFI_NOT_FOUND;
+  }
+
+  MaxLength = MIN (AsciiStrLen (FormatString), OC_LOG_PREFIX_CHAR_MAX);
+  for (Index = 1; Index < MaxLength; ++Index) {
+    Curr = FormatString[Index];
+
+    //
+    // Match the first occurrence of colon.
+    //
+    if (Curr == ':') {
+      break;
+    }
+
+    //
+    // Except for colon, a valid prefix must be either 0-9, or uppercase letter.
+    //
+    if (!(IsAsciiNumber (Curr) || (Curr >= 'A' && Curr <= 'Z'))) {
+      return EFI_NOT_FOUND;
+    }
+  }
+
+  //
+  // If Index went through the end, then ':' was not found.
+  //
+  if (Index == MaxLength) {
+    return EFI_NOT_FOUND;
+  }
+
+  CopyMem (Prefix, FormatString, Index);
+  Prefix[Index] = '\0';
+  return EFI_SUCCESS;
+}
+
+STATIC
+BOOLEAN
+IsPrefixFiltered (
+  IN   CONST CHAR8          *FormatString,
+  IN   CONST OC_FLEX_ARRAY  *FlexFilters,
+  OUT  OC_LOG_FILTER_MODE   *FilterMode
+  )
+{
+  UINTN       Index;
+  CHAR8       Prefix[OC_LOG_PREFIX_CHAR_MAX + 1];
+  EFI_STATUS  Status;
+  CHAR8       **Value;
+
+  ASSERT (FormatString != NULL);
+  ASSERT (FilterMode != NULL);
+
+  if (FlexFilters == NULL) {
+    *FilterMode = OcLogFilterModeNoFilter;
+    return FALSE;
+  }
+
+  Status = GetLogPrefix (FormatString, Prefix);
+  if (EFI_ERROR (Status)) {
+    *FilterMode = OcLogFilterModeNoFilter;
+    return FALSE;
+  }
+
+  Value = (CHAR8 **) OcFlexArrayItemAt (FlexFilters, 0);
+  ASSERT (Value != NULL);
+  //
+  // One mere symbol? This is borked. 
+  //
+  if (AsciiStrLen (*Value) <= 1) {
+    *FilterMode = OcLogFilterModeNoFilter;
+    return FALSE;
+  }
+  //
+  // Determine filter mode and process Index 0.
+  //
+  if ((*Value)[0] == '+') {
```

Comment:
Define `CONST CHAR8 *FilterList = *Value;` then use it. Increment when you see `+` or `-` in the beginning.

Reply:
Resolved by [directly parsing `LogModule`](https://github.com/acidanthera/OpenCorePkg/pull/322/commits/8a869c21e0819e132581bc7e4ec66a3b87ec70d0#diff-d5e24c55a7bdcf0fbd974a2e7ca85edbebb2ff4cfd00ca0cc31e5c23fe8051fdR653).